### PR TITLE
[Gardening] Silence Some Warning Spew

### PR DESCRIPTION
**Pull request by CodaFi**
The linux builders seem to have more warning options on than the macOS builders.  Silence some warnings it raises.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please test

File: lib/AST/IndexSubset.cpp [hunk 1]
```diff
@@ -16,7 +16,6 @@ using namespace swift;
 
 IndexSubset *
 IndexSubset::getFromString(ASTContext &ctx, StringRef string) {
-  if (string.size() < 0) return nullptr;
```
Comment (CodaFi) [inline lib/AST/IndexSubset.cpp]: @dan-zheng This seems like a typo. Should this have been `string.empty()`?

Comment (dan-zheng) [reply lib/AST/IndexSubset.cpp]: Yes, I think `if (string.empty()) return nullptr` was intended here. Thanks for the catch!

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
Hm, have to go update that unit test then.

**Review by slavapestov — APPROVED** [review]
Thanks!

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
:sail

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Comment by CodaFi** [comment]
Linux failure is unrelated

@swift-ci please smoke test Linux platform

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**